src/main/java/com/example/springbookstore/service/BookServiceImpl.java

```diff
@@ -1,27 +1,84 @@
 package com.example.springbookstore.service;
 
+import com.example.springbookstore.dto.BookDto;
+import com.example.springbookstore.dto.CreateBookRequestDto;
 import com.example.springbookstore.entity.Book;
+import com.example.springbookstore.exception.EntityNotFoundException;
+import com.example.springbookstore.mapper.BookMapper;
 import com.example.springbookstore.repository.BookRepository;
 import java.util.List;
+import java.util.stream.Collectors;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class BookServiceImpl implements BookService {
+
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
         this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
+    }
+
+    @Override
+    public BookDto createBook(CreateBookRequestDto createBookRequestDto) {
+        Book book = bookMapper.createRequestDtoToEntity(createBookRequestDto);
+        Book savedBook = bookRepository.save(book);
+        return bookMapper.entityToDto(savedBook);
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> getAllBooks() {
+        List<Book> books = bookRepository.findAll();
+        return books.stream()
+                .map(bookMapper::entityToDto)
+                .collect(Collectors.toList());
```

Comment:
you use java 17 so .toList() would be more logical to see instead of .collect(Collectors.toList())